pandas/io/formats/csvs.py

```diff
@@ -252,22 +255,58 @@ def save(self) -> None:
             storage_options=self.storage_options,
         ) as handles:
             # Note: self.encoding is irrelevant here
+            self._save(handles.handle)
+
+    def _save_pyarrow(self, handle) -> None:
+        pa = import_optional_dependency("pyarrow")
+        pa_csv = import_optional_dependency("pyarrow.csv")
+        # Convert index to column and rename name to empty string
+        # since we serialize the index as basically a column with no name
+        # TODO: this won't work for multi-indexes
+        obj = self.obj.reset_index(names=[""])
+
+        table = pa.Table.from_pandas(obj)
+
+        # Map quoting arg to pyarrow equivalents
+        pa_quoting = None
+        if self.quoting == csvlib.QUOTE_MINIMAL:
+            pa_quoting = "needed"
+        elif self.quoting == csvlib.QUOTE_ALL:
+            # TODO: Is this a 1-1 mapping?
+            # This doesn't quote nulls, check if Python does this
+            pa_quoting = "all_valid"
+        elif self.quoting == csvlib.QUOTE_NONE:
+            pa_quoting = "none"
+        else:
+            raise ValueError(
+                f"Quoting option {self.quoting} is not supported with engine='pyarrow'"
+            )
+
+        write_options = pa_csv.WriteOptions(
+            include_header=self._need_to_save_header,
+            batch_size=self.chunksize,
+            delimiter=self.sep,
+            quoting_style=pa_quoting,
+        )
+        # pa_csv.write_csv(table, handle, write_options)
+        pa_csv.write_csv(table, self.filepath_or_buffer, write_options)
```

Comment:
Haven't decided yet whether to let pyarrow figure out how to do file handling on its own or use ``get_handle``.

``get_handle`` doesn't work out of the box due to pyarrow only working on binary handles IIRC.

Reply:
Would it be possible to try to convert the handle to a binary one?

Reply:
How to set specific encoding for pyarrow's `write_csv`?